app/models/merge_request.rb

```diff
@@ -98,4 +98,43 @@ def organisations_without_schemes
 
     ([absorbing_organisation] + merging_organisations).reject(&:has_visible_schemes?)
   end
+
+  def filter_relationships(absorbing_relationships, merging_relationships, absorbing_organisation, merging_organisations)
+    filtered_absorbing_relationships = absorbing_relationships.reject do |relationship|
+      merging_relationships.include?(relationship) || merging_organisations.include?(relationship)
+    end
+
+    filtered_merging_relationships = merging_relationships.reject do |relationship|
+      absorbing_relationships.include?(relationship) || relationship == absorbing_organisation || merging_organisations.include?(relationship)
+    end
+
+    (filtered_absorbing_relationships + filtered_merging_relationships).uniq
+  end
```

Comment:
This method filters out relationships:

- from absorbing orgs (absorbing_relationships) that are also present in merging orgs (merging_relationships)
- from merging orgs (merging_relationships) that are also present in absorbing orgs (absorbing_relationships)
- combines those filtered lists and removes duplicates

This method is used for both stock_owners and managing_agents